sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -513,6 +513,43 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
     }
   }
 
+  test("Save csv with custom charset") {
+    Seq("iso-8859-1", "utf-8", "windows-1250").foreach { encoding =>
+      withTempDir { dir =>
+        val csvDir = new File(dir, "csv").getCanonicalPath
+        // scalastyle:off
+        val originalDF = Seq("µß áâä ÁÂÄ").toDF("_c0")
+        // scalastyle:on
+        originalDF.write
+          .option("header", "false")
+          .option("encoding", encoding)
+          .csv(csvDir)
+
+        val df = spark
+          .read
+          .option("header", "false")
+          .option("encoding", encoding)
```

Comment:
I think our CSV read encoding option is incomplete for now .. there are many discussions about this now. I am going to fix the read path soon. Let me revisit this after fixing it.

Reply:
Now it's fine. I think we decided to support encoding in CSV/JSON datasources. Ignore the comment above. We can proceed separately.